sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -130,6 +130,14 @@ case class Join(
   }
 }
 
+/**
+ * A hint for the optimizer that we should broadcast the `child` if used in a join operator.
+ */
+case class BroadcastHint(child: LogicalPlan) extends UnaryNode {
```

Comment:
I'm thinking about this more and actually maybe what we want to do is have this specific node, and a patter that recognizes `canBroadcast`.  This pattern can check either for a small enough size or this hint and we can use that anywhere we are planning a broadcast operator.  The reasoning is that messing with statistics could have other consequences.